Tests/UnitTests/FileShareService.DesktopClientTests/Modules/Search/BatchDetailsViewModelTest.cs

```diff
@@ -0,0 +1,100 @@
+            fakesaveFileDialogService = A.Fake<ISaveFileDialogService>();
+        }
+        [Test]
+        public async Task TestDownloadFileWithFileExistsAndYesResponse()
+        {
+           
+            var BatchDetailVM = new BatchDetailsViewModel(fakeFileShareApiAdminClientFactory, fakeMessageBoxService, fakeFileService , fakesaveFileDialogService);
+            
+            BatchDetailVM.Files = new List<BatchDetailsFiles>() { new BatchDetailsFiles() { Filename = "AFilename.txt", FileSize = 100 } };
+            A.CallTo(() => fakesaveFileDialogService.SaveFileDialog(A<string>.Ignored)).Returns(Path.Combine(Environment.CurrentDirectory));
+
+            BatchDetailVM.BatchId = Guid.NewGuid().ToString();
+            string fileDownloadPath = Path.Combine(Environment.CurrentDirectory, "AFilename.txt");          
+            A.CallTo(() => fakeFileService.Exists(fileDownloadPath)).Returns(true);
+            A.CallTo(() => fakeMessageBoxService.ShowMessageBox(A<string>.Ignored, A<string>.Ignored,MessageBoxButton.YesNo, A<MessageBoxImage>.Ignored)).Returns(MessageBoxResult.Yes);
+            A.CallTo(() => fakeFileShareApiAdminClientFactory.Build()).Returns(fakeFileShareApiAdminClient);
+            A.CallTo(() => fakeFileShareApiAdminClient.DownloadFileAsync(A<string>.Ignored, A<string>.Ignored, A<FileStream>.Ignored, A<long>.Ignored, A<CancellationToken>.Ignored)).Returns(new Result<DownloadFileResponse> { IsSuccess = true, StatusCode = 206 });
+
+            BatchDetailVM.DownloadExecutionCommand.Execute("AFilename.txt");
+
+            A.CallTo(() => fakeMessageBoxService.ShowMessageBox(A<string>.Ignored, A<string>.Ignored, MessageBoxButton.OK, A<MessageBoxImage>.Ignored)).MustHaveHappened();
+        } 
+
+        [Test]
+        public async Task TestDownloadFileWithFileExistsAndNoResponse()
+        {
+            var BatchDetailVM = new BatchDetailsViewModel(fakeFileShareApiAdminClientFactory, fakeMessageBoxService, fakeFileService, fakesaveFileDialogService);
+
+            BatchDetailVM.Files = new List<BatchDetailsFiles>() { new BatchDetailsFiles() { Filename = "AFilename.txt", FileSize = 100 } };
+            A.CallTo(() => fakesaveFileDialogService.SaveFileDialog(A<string>.Ignored)).Returns(Path.Combine(Environment.CurrentDirectory));
+
+            BatchDetailVM.BatchId = Guid.NewGuid().ToString();
+            string fileDownloadPath = Path.Combine(Environment.CurrentDirectory, "AFilename.txt");
+            A.CallTo(() => fakeFileService.Exists(fileDownloadPath)).Returns(true);
+            A.CallTo(() => fakeMessageBoxService.ShowMessageBox(A<string>.Ignored, A<string>.Ignored, MessageBoxButton.YesNo, A<MessageBoxImage>.Ignored)).Returns(MessageBoxResult.No);
+
+            BatchDetailVM.DownloadExecutionCommand.Execute("AFilename.txt");
+
+            A.CallTo(() => fakeFileShareApiAdminClientFactory.Build()).Returns(fakeFileShareApiAdminClient);
```

Comment:
Please change to a MustNotHaveHappened() assertion
```suggestion
            A.CallTo(() => fakeFileShareApiAdminClientFactory.Build()).MustNotHaveHappened();
```

Reply:
resolved